File: number_subarray_maximum.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends


class Solution{
    public:
    long countSubarrays(int a[], int n, int L, int R)
    {
        // Complete the function
        long btwr=0,lesl=0,ans=0;
        for(int i=0;i<n;i++){
            if(a[i]<L){
                ans+=btwr;
                lesl++;
            }else if(a[i]>R){
                btwr=lesl=0;
            }else{
                ans+=btwr+lesl+1;
                btwr+=lesl+1;
                lesl=0;
            }
        }
        return ans;
    }
};

//{ Driver Code Starts.
// driver program
int main()
{
    int t;
    cin >> t;
    while(t--)
    {
        
        int n ,l,r;
        cin >> n>>l>>r;
        int a[n];
        for(int i=0;i<n;i++)
        cin >> a[i];
        Solution ob;
    	cout << ob.countSubarrays(a, n, l, r)<<endl;
    }
	return 0;
}

// } Driver Code Ends

File: number_subarray_maximum.py
#User function Template for python3

class Solution:
    def countSubarrays(self, a,n,L,R): 
        # Complete the function
        btlr,ans,lessl=0,0,0
        for i in range(n):
            if a[i]<L:
                ans+=btlr
                lessl+=1
            elif a[i]>R:
                btlr,lessl=0,0
            else:
                ans+=btlr+lessl+1
                btlr+=lessl+1
                lessl=0
        return ans        


#{ 
 # Driver Code Starts
#Initial Template for Python 3



for _ in range(0,int(input())):
    n,l,r = map(int, input().strip().split())
    arr = list(map(int,input().strip().split()))
    ob = Solution()
    v = ob.countSubarrays(arr, n, l, r)
    print(v)
    
# } Driver Code Ends